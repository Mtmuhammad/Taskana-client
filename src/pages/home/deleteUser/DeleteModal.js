// Modal to delete a user

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import "../../projects/projectCard/ProjectCard.scss";
import React from "react";

const DeleteModal = ({ users, user, setUsers, setSuccess, setErrMsg }) => {
  const axiosPrivate = useAxiosPrivate();

  // handle form submission for user deletion
  const handleDelete = async (e) => {
    e.preventDefault();
    const id = user?.empNumber;
    try {
      await axiosPrivate.delete(`/users/${id}`, {
        headers: { "Content-Type": "application/json" },
        withCredentials: true,
      });

      setUsers(users.filter((u) => u.empNumber !== id));
      setSuccess("User Deleted!");
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }
  };

  return (
    <div
      className="modal fade"
      id={`deleteUser${user?.empNumber}`}
      tabIndex="-1"
      aria-hidden="true"
      style={{ display: "none", color: "black" }}
      data-testid="delete-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="deleteUserLabel"
            >
              {" "}
              Delete item Permanently?
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body justify-content-center flex-column d-flex">
            <i
              data-testid="delete-icon"
              className="bx bxs-trash text-danger display-2 text-center mt-2"
            ></i>
            <p data-testid="delete-text" className="mt-4 fs-3 text-center">
              You can only delete this item Permanently
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleDelete}
              type="button"
              className="fs-5 btn btn-danger color-fff"
              data-bs-dismiss="modal"
              data-testid="delete-btn"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
