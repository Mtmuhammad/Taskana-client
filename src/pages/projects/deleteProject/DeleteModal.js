// Modal to delete a new project

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import "../projectCard/ProjectCard.scss";

const DeleteModal = ({
  setProjects,
  projects,
  project,
  setSuccess,
  setErrMsg,
}) => {
  const axiosPrivate = useAxiosPrivate();

  // handle form submission for project deletion
  const handleDelete = async (e) => {
    e.preventDefault();
    try {
      const id = project.id;
      await axiosPrivate.delete(`/projects/${id}`, {
        headers: { "Content-Type": "application/json" },
        withCredentials: true,
      });

      setProjects(projects.filter((p) => p.id !== id));
      setSuccess("Project Deleted!");
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }
  };

  return (
    <div
      className="modal fade"
      id={`deleteProject${project?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      style={{ display: "none" }}
      data-testid="delete-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="deleteProjectLabel"
            >
              {" "}
              Delete item Permanently?
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body justify-content-center flex-column d-flex">
            <i
              data-testid="delete-icon"
              className="bx bxs-trash text-danger display-2 text-center mt-2"
            ></i>
            <p data-testid="delete-text" className="mt-4 fs-3 text-center">
              You can only delete this item Permanently
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleDelete}
              type="button"
              className="fs-5 btn btn-danger color-fff"
              data-bs-dismiss="modal"
              data-testid="delete-btn"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
