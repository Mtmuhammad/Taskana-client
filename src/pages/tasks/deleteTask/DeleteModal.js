// Modal to delete a new task

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import useAuth from "../../../hooks/useAuth";

const DeleteModal = ({
  setShowTasks,
  filterTasks,
  setUserTasks,
  task,
  setSuccess,
  setErrMsg,
}) => {
  const { auth } = useAuth();
  const axiosPrivate = useAxiosPrivate();

  // handle form submission for project deletion
  const handleDelete = async (e) => {
    e.preventDefault();
    try {
      let isMounted = true;
      const controller = new AbortController();
      const id = task.id;
      await axiosPrivate.delete(`/tasks/${id}`, {
        headers: { "Content-Type": "application/json" },
        withCredentials: true,
      });

      const res = await axiosPrivate.get(`/tasks/${auth?.user?.empNumber}`, {
        signal: controller.signal,
      });
      isMounted && setUserTasks(res.data.tasks);
      setShowTasks(res.data.tasks);
      filterTasks(res);
      setSuccess("Task Deleted!");
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }
  };

  return (
    <div
      className="modal fade"
      id={`deleteTask${task?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      style={{ display: "none", color: "black" }}
      data-testid="delete-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="deleteProjectLabel"
            >
              {" "}
              Delete item Permanently?
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body justify-content-center flex-column d-flex">
            <i
              data-testid="delete-icon"
              className="bx bxs-trash text-danger display-2 text-center mt-2"
            ></i>
            <p data-testid="delete-text" className="mt-4 fs-3 text-center">
              You can only delete this item Permanently
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleDelete}
              type="button"
              className="fs-5 btn btn-danger color-fff"
              data-bs-dismiss="modal"
              data-testid="delete-btn"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
