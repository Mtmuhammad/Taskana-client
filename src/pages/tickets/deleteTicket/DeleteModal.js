// Modal to delete a new ticket

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import useAuth from "../../../hooks/useAuth";
import React from 'react'

const DeleteModal = ({
  setShowTickets,
  setAssignedTickets,
  setIncompleteTickets,
  setOpenTickets,
  setAllTickets,
  setIsLoading,
  ticket,
  setSuccess,
  setErrMsg,
}) => {
  const { auth } = useAuth();
  const axiosPrivate = useAxiosPrivate();

  // handle form submission for ticket deletion
  const handleDelete = async (e) => {
    e.preventDefault();
    try {
      let isMounted = true;
      const controller = new AbortController();
      const id = ticket.id;
      await axiosPrivate.delete(`/tickets/${id}`, {
        headers: { "Content-Type": "application/json" },
        withCredentials: true,
      });

      const res = await axiosPrivate.get(`/tickets`, {
        signal: controller.signal,
      });

      let tickets = res.data.tickets

      isMounted && setAllTickets(tickets);
      setShowTickets();
      setShowTickets(tickets);
      setIsLoading(false);

      setAssignedTickets(
        tickets.filter(
          (ticket) => ticket.assignedTo === auth?.user?.empNumber
        ) || 0
      );
      setIncompleteTickets(
        tickets.filter((ticket) => ticket.status === "In Progress") || 0
      );
      setOpenTickets(
        tickets.filter((ticket) => ticket.status === "Open") || 0
      );

      setErrMsg("");
      setSuccess("Ticket Deleted!");
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }
  };

  return (
    <div
      className="modal fade"
      id={`deleteTicket${ticket?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      style={{ display: "none", color: "black" }}
      data-testid="delete-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="deleteProjectLabel"
            >
              {" "}
              Delete item Permanently?
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body justify-content-center flex-column d-flex">
            <i
              data-testid="delete-icon"
              className="bx bxs-trash text-danger display-2 text-center mt-2"
            ></i>
            <p data-testid="delete-text" className="mt-4 fs-3 text-center">
              You can only delete this item Permanently
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleDelete}
              type="button"
              className="fs-5 btn btn-danger color-fff"
              data-bs-dismiss="modal"
              data-testid="delete-btn"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
