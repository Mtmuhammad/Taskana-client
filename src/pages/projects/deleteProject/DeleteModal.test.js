import React from "react";
import { render, screen } from "@testing-library/react";
import DeleteModal from "./DeleteModal";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<DeleteModal />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("create modal", () => {
  test("should render modal", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const modal = screen.getByTestId("delete-modal");
    expect(modal).toBeInTheDocument();
    expect(modal).toHaveClass("modal fade");
  });
  test("should render modal header", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const header = screen.getByTestId("header");
    expect(header).toBeInTheDocument();
    expect(header).toHaveClass("modal-title");
    expect(header).toContainHTML("Delete item Permanently?");
  });
  test("should render close btn (x)", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const close = screen.getByTestId("close-btn");
    expect(close).toBeInTheDocument();
    expect(close).toHaveClass("btn-close");
  });
  test("should render delete icon", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const icon = screen.getByTestId("delete-icon");
    expect(icon).toHaveClass("bx bxs-trash text-danger");
  });
  test("should render delete text", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const text = screen.getByTestId("delete-text");
    expect(text).toHaveClass("text-center");
  });
  test("should render cancel button", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("cancel-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-secondary");
  });
  test("should render create button", () => {
    render(<DeleteModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("delete-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-danger");
  });
});
