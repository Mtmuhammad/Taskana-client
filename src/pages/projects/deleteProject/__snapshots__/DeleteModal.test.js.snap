// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="modal fade"
    data-testid="delete-modal"
    id="deleteProjectundefined"
    style="display: none;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title  fw-bold"
            data-testid="header"
            id="deleteProjectLabel"
          >
             Delete item Permanently?
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            data-testid="close-btn"
            type="button"
          />
        </div>
        <div
          class="modal-body justify-content-center flex-column d-flex"
        >
          <i
            class="bx bxs-trash text-danger display-2 text-center mt-2"
            data-testid="delete-icon"
          />
          <p
            class="mt-4 fs-3 text-center"
            data-testid="delete-text"
          >
            You can only delete this item Permanently
          </p>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="fs-5 btn btn-secondary"
            data-bs-dismiss="modal"
            data-testid="cancel-btn"
            type="button"
          >
            Cancel
          </button>
          <button
            class="fs-5 btn btn-danger color-fff"
            data-bs-dismiss="modal"
            data-testid="delete-btn"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
